api/base/utils.py

```diff
@@ -147,18 +147,15 @@ def default_node_list_queryset(model_cls):
     assert model_cls in {Node, Registration}
     return model_cls.objects.filter(is_deleted=False).annotate(region=F('addons_osfstorage_node_settings__region___id'))
 
-def default_node_permission_queryset(user, model_cls):
+def default_node_permission_queryset(auth, model_cls):
     assert model_cls in {Node, Registration}
-    if user is None or user.is_anonymous:
-        return model_cls.objects.filter(is_public=True)
-    sub_qs = Contributor.objects.filter(node=OuterRef('pk'), user__id=user.id, read=True)
-    return model_cls.objects.annotate(contrib=Exists(sub_qs)).filter(Q(contrib=True) | Q(is_public=True))
+    return model_cls.objects.can_view(auth.user, auth.private_link)
```

Comment:
Taking advantage of `can_view` which takes care of all the pieces we need - adds in public nodes, nodes you can view through private links, and/or have read perms on.